src/datachain/config.py

```diff
@@ -1,62 +1,131 @@
+
+        return merged_conf
+
+    def read(self) -> TOMLDocument:
+        if self.level is None:
+            return self.load_config_to_level()
+        return self.load_one(self.level)
+
+    @contextmanager
+    def edit(self):
+        config = self.load_one(self.level)
+        yield config
+
+        self.write(config)
+
+    def config_file(self):
+        return DataChainDir(self.get_dir(self.level)).config
+
+    def write(self, config: TOMLDocument):
+        with open(self.config_file(), "w") as f:
+            dump(config, f)
+
+    def get_remote_config(self, remote: str = "") -> Mapping[str, str]:
+        config = self.read()
+
+        if not config:
             return {"type": "local"}
-    try:
-        remote_conf: Mapping[str, str] = config["remote"][remote]  # type: ignore[assignment,index]
-    except KeyError:
-        raise Exception(
-            f"missing config section for default remote: remote.{remote}"
-        ) from None
-    except Exception as exc:
-        raise Exception("invalid config") from exc
-
-    if not isinstance(remote_conf, Mapping):
-        raise TypeError(f"config section remote.{remote} must be a mapping")
-
-    remote_type = remote_conf.get("type")
-    if remote_type not in ("local", "http"):
-        raise Exception(
-            f'config section remote.{remote} must have "type" with one of: '
-            '"local", "http"'
-        )
-
-    if remote_type == "http":
-        for key in ["url", "username", "token"]:
+        if not remote:
             try:
-                remote_conf[key]
+                remote = config["core"]["default-remote"]  # type: ignore[index,assignment]
             except KeyError:
-                raise Exception(
-                    f"config section remote.{remote} of type {remote_type} "
-                    f"must contain key {key}"
-                ) from None
-    elif remote_type != "local":
-        raise Exception(
-            f"config section remote.{remote} has invalid remote type {remote_type}"
-        )
-    return remote_conf
+                return {"type": "local"}
+        try:
+            remote_conf: Mapping[str, str] = config["remote"][remote]  # type: ignore[assignment,index]
+        except KeyError:
+            raise Exception(
+                f"missing config section for default remote: remote.{remote}"
+            ) from None
+        except Exception as exc:
+            raise Exception("invalid config") from exc
+
+        if not isinstance(remote_conf, Mapping):
+            raise TypeError(f"config section remote.{remote} must be a mapping")
+
+        remote_type = remote_conf.get("type")
+        if remote_type not in ("local", "http"):
+            raise Exception(
+                f'config section remote.{remote} must have "type" with one of: '
+                '"local", "http"'
+            )
+
+        if remote_type == "http":
+            for key in ["url", "username", "token"]:
```

Comment:
Just wonder, should we also have `team_id` as a part of remote config? User can be a member of many teams in Studio 🤔 Or `team_id` will be required as a (required? optional?) param in API calls?

Reply:
This may be part of future changes when implementing some functionality with the token access.